### ISSUE TITLE
Android getInitialPushPayload() always resolves to null (7.0.0-beta)

### ISSUE THREAD

**Issue by webdesignpresse** [andrehtissot/cordova-plugin-fcm-with-dependecy-updated#103]
Hi,

we are currently using the beta of your plugin. On iOS the onNotification method is triggered when the app is in foreground as well as in background. On Android unfortunately it only triggers the onNotification method when the app is on foreground. In your documentation you mentioned the getInitialPushPayload() method, but this always resolves to null.

```
const ppl = await window.FCM.getInitialPushPayload();
console.log(ppl) // null

// in foreground and background on iOS
// only in foreground in Android
window.FCM.onNotification(payload => console.log(payload)) // Notification object
```

**Comment by danielburger1337** [comment]
I don't necessarily know if this is an issue or misuse, since this is contradicting to other open issues, but I have the exact same issue.

In your documentation, it is mentioned that the onNotification-Eventlistener no longer gets retriggered when you open the closed app. 


On iOS this is not the case. The handler gets executed every time like in previous versions (imo the preferred way). 

On Android the handler only gets executed only while the app is opened (in foregroud). I can not get any data with the getInitalPushPayload-Method. I basically tried, while USB-Debugging, to execute the method in the console right after I opened the app via a push notification.

Used Version: 7.0.0-beta.0

Thank you for your help, time and commitment to this awesome project.

**Comment by andrehtissot** [comment]
@webdesignpresse usually this behavior is result of missing the `click_action` key as `"FCM_PLUGIN_ACTIVITY"`, like presented in https://github.com/andrehtissot/cordova-plugin-fcm-with-dependecy-updated#send-notification-payload-example-rest-api

**Comment by andrehtissot** [comment]
@danielburger1337

> In your documentation, it is mentioned that the onNotification-Eventlistener no longer gets retriggered when you open the closed app.

I'm sorry if wasn't clear, I'll try to improve that section of the documentation to avoid confusion. Thank you for the message.

What I mean it has: If the subscription to notification events happens after the notification has been fired, it'll be lost. As it is expected that you'd not always be able to catch the notification tapping the opened the app, the `FCM.getInitalPushPayload` method was introduced.
 
> On iOS this is not the case. The handler gets executed every time like in previous versions (imo the preferred way).

On iOS, there is a small delay to Plugin's initialization. Because of it, if you are able to subscribe within this time window, the event will be cached.

> On Android the handler only gets executed only while the app is opened (in foregroud). I can not get any data with the getInitalPushPayload-Method. I basically tried, while USB-Debugging, to execute the method in the console right after I opened the app via a push notification.

If the push payload contains the "notification" key, you have to add the `click_action` key as `"FCM_PLUGIN_ACTIVITY"`, like presented in https://github.com/andrehtissot/cordova-plugin-fcm-with-dependecy-updated#send-notification-payload-example-rest-api
Without this, the Android OS won't send the payload to the background service that triggers the JS context.

> Used Version: 7.0.0-beta.0 & .2

Thank you letting me know.

> Thank you for your help, time and commitment to this awesome project.

Thank you for being part of this awesome community.

**Comment by danielburger1337** [comment]
@andrehtissot 
Thank you very much for your quick and informing reply.

I think I successfully traced back my issue to when I migratated to the new Firebase HTTP v1 API on the server side of things which coincided with upgrading your plugin to the newest BETA version.

I will leave my "findings" here, in case anyone else comes upon the same problem as I have.

In their [documentation](https://firebase.google.com/docs/cloud-messaging/migrate-v1?authuser=0#example-simple-notification-message) it is not totally obvious what a click_action actually does.
While testing, I always received a 400 Bad Request because of the Property "click_action" in the notification object. So, naturally, i removed it and thought that it was deprecated in this newer version of the api to the extend that I was using it for (this Plugin for cordova). Only while integrating web push into the message object, I realized that there were major platform differences and this specific click_action has to be present in the android object. Basically my message object now looks like this: (for my use case I don't need any specific aps (iOS) options)


```
"message": {
  "token": ...,
  "android": {
    "notification": {
      "click_action": "FCM_PLUGIN_ACTIVITY"
    }
  },
  "web_push": {
    "notification": {
      "icon": "https://example.com/logo.png"
    },
    "fcm_options": {
      "link": "https://example.com/link-to-follow"
     }
   },
   "notification": {
     "body": "The body of my message",
     "title": "The title of my message"
   },
   "data": {
      "property1": "data1",
       ....
    } 
}
```
To my understanding it looks like on Android, the background service has to invoke a certain function provided by the app in order to have access to the message payload. So I think I am correct in guessing that your plugin is providing this FCM_PLUGIN_ACTIVITY function / method / whatever which then in turn brings it to the Javascript context.

I think it would be helpful if you provide a little more information as to why this click_action has to exist and what it does in [your documentation](https://github.com/andrehtissot/cordova-plugin-fcm-with-dependecy-updated#send-notification-payload-example-rest-api) like described above. 

Thank your very much for your patience and excellent work. I hope that my experience helps anyone who has a similar issue.
